### PR TITLE
wasm32-wasi -> wasm32-wasip1

### PR DESCRIPTION
**Pull request by dp-mason**
changed the default target from deprecated wasm32-wasi target to wasm32-wasip1. Added rustup target check to prepare.sh

### CODE REVIEW

File: template/prepare.sh [hunk 1]
```diff
@@ -28,6 +28,21 @@ if ! (command_exists cargo); then
   echo ""
 fi
 
+if ! (command_exists rustup); then
+  missing_deps=1
+  echo "❌ rustup is missing. Check your rust installation."
```
Comment (bhelx) [inline template/prepare.sh:33]: Note: not a blocker, but we should come back and add some links here about how to install rustup.

File: template/prepare.sh [hunk 2]
```diff
@@ -28,6 +28,21 @@ if ! (command_exists cargo); then
...
+  echo ""
+elif ! (rustup target list --installed | grep -q '^wasm32-wasip1$'); then
+  if ! (rustup target add wasm32-wasip1); then
```
Comment (bhelx) [inline template/prepare.sh]: Also not a blocker, but we should install the unknown target as well

**Review by bhelx — APPROVED** [review]
Much appreciated!

**Comment by dp-mason** [comment]
> Much appreciated!

No prob!

**Pull request merged**